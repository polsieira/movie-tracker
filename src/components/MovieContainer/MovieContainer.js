import React from 'react';
import './MovieContainer.scss';
import MovieCard from '../MovieCard/MovieCard';
import NavigationBar from '../NavigationBar/NavigationBar';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';

const MovieContainer = ({ movies, error }) => {
  const displayMovies = movies.map(movie => {
    return (
      <MovieCard
        {...movie}
        key={movie.id}
      />
    )
  })

  return (
    <>
      <NavigationBar />
      {error && <p className='errorMsg'>Unable to load movies. We hate movies. Go to IMDB.</p>}
      <div className='MovieContainer'>
        {displayMovies}
      </div>
    </>
  )

}

const mapStateToProps = ({ movies, error }) => ({
  movies: movies,
  error
});

export default connect(mapStateToProps)(MovieContainer);
