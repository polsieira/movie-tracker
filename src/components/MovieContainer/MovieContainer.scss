.MovieContainer {
  display: grid;
  gap: 20px;
  grid-template-columns: 1fr 1fr;
  padding: 150px 10%;
}

@media only screen and (max-width: 900px) {
  .MovieContainer {
    gap: 20px;
    grid-template-columns: 1fr;
  }
}