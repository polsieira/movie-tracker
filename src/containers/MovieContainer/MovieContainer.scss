.MovieContainer {
  display: grid;
  gap: 20px;
  min-height: 100vh;
  grid-template-columns: 1fr 1fr;
  padding: 150px 10%;
}

.errorMsg {
  color: black;
  font-size: 36px;
  margin: 0px;
  padding: 200px;
  text-align: center;
}

@media only screen and (max-width: 900px) {
  .MovieContainer {
    gap: 20px;
    grid-template-columns: 1fr;
  }
}