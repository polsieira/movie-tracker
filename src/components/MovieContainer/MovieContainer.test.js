import { MovieContainer, mapStateToProps } from './MovieContainer';
import { shallow } from 'enzyme';
import React from 'react';


describe('MovieContainer', () => {
  describe('MovieContainer component', () => {
    let wrapper;
    const mockHandleFavorite = jest.fn();
    beforeEach(() => {
      wrapper = shallow(<MovieContainer
        movies={[{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          id: 6
        }]}
        errorMsg=''
        handleFavorite={mockHandleFavorite}
      />)
    })

    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should rende')
  });

  describe('mapStateToProps', () => {
    it('should return an object with the todos array', () => {
      // Setup
      const mockState = {
        movies: [{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          movie_id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          movie_id: 6
        }],
        user: {
          name: 'Pol',
          id: 2,
          isSignedIn: true
        },
        errorMsg: ''

      };
      const expected = {
        movies: [{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          movie_id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          movie_id: 6
        }],
        errorMsg: ''
      };

      // Execution
      const mappedProps = mapStateToProps(mockState);

      // Expectation
      expect(mappedProps).toEqual(expected);
    });
  });

})