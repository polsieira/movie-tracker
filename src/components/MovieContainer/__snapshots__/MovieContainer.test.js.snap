// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer MovieContainer component should match a snapshot 1`] = `
<Fragment>
  <Connect(NavigationBar) />
  <div
    className="MovieContainer"
  >
    <Connect(MovieCard)
      average_rating={6.5}
      handleFavorite={[MockFunction]}
      id={5}
      key="5"
      movie={
        Object {
          "average_rating": 6.5,
          "id": 5,
          "popularity": 55.55,
          "release_date": "2017",
          "title": "Wicked",
        }
      }
      popularity={55.55}
      release_date="2017"
      title="Wicked"
    />
    <Connect(MovieCard)
      average_rating={7.5}
      handleFavorite={[MockFunction]}
      id={6}
      key="6"
      movie={
        Object {
          "average_rating": 7.5,
          "id": 6,
          "popularity": 68.55,
          "release_date": "2019",
          "title": "Joker",
        }
      }
      popularity={68.55}
      release_date="2019"
      title="Joker"
    />
  </div>
</Fragment>
`;

exports[`MovieContainer MovieContainer component should render an error message 1`] = `
<Fragment>
  <Connect(NavigationBar) />
  <p
    className="errorMsg"
  >
    Unable to load movies. We hate movies. Go to IMDB.
  </p>
  <div
    className="MovieContainer"
  >
    <Connect(MovieCard)
      average_rating={6.5}
      handleFavorite={[MockFunction]}
      id={5}
      key="5"
      movie={
        Object {
          "average_rating": 6.5,
          "id": 5,
          "popularity": 55.55,
          "release_date": "2017",
          "title": "Wicked",
        }
      }
      popularity={55.55}
      release_date="2017"
      title="Wicked"
    />
    <Connect(MovieCard)
      average_rating={7.5}
      handleFavorite={[MockFunction]}
      id={6}
      key="6"
      movie={
        Object {
          "average_rating": 7.5,
          "id": 6,
          "popularity": 68.55,
          "release_date": "2019",
          "title": "Joker",
        }
      }
      popularity={68.55}
      release_date="2019"
      title="Joker"
    />
  </div>
</Fragment>
`;
